### PR TITLE
Feat/search users

### PR DESCRIPTION
**Pull request by binnmti**
## Summary by CodeRabbit

- **New Features**
  - Introduced functions for querying edit and view history data.
  - Added search functionality for users by ID.

- **Enhancements**
  - `Menu` component now supports custom styling via `className` prop.
  - Improved `UserSelectMenu` for better user selection and search experience.

- **Bug Fixes**
  - Removed unused imports and props in `Edit` page.

- **Refactor**
  - Renamed `Usertem.tsx` to `UserItem.tsx`.
  - Updated `userConverter` function to exclude `id` field from validation.

- **Chores**
  - Added debugging configuration for VSCode.
  - Registered new services and repositories in dependency injection setup.

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    "configurations": [
+        {
+            "type": "chrome",
+            "request": "launch",
+            "name": "Launch Chrome against localhost",
+            "url": "http://localhost:5173",
+            "webRoot": "${workspaceFolder}"
+        }
+    ]
+}
```
Comment (Harineko0) [inline .vscode/launch.json:15]: Is this file necessary? You might accidentally added this?

File: one-pager-maker/src/entity/user/userType.ts [hunk 2]
```diff
@@ -17,7 +17,7 @@ export type User = z.infer<typeof userSchema>;
 
 export const userConverter: FirestoreDataConverter<User> = {
     fromFirestore(snapshot): User {
-        const data = {...snapshot.data(), id: snapshot.id};
+        const data = {...snapshot.data()};
```
Comment (Harineko0) [inline one-pager-maker/src/entity/user/userType.ts:20]: Please do not remove `id: snapshot.id`!

**Review by Harineko0 — CHANGES_REQUESTED** [review]
LGTM! bu I requested 1 change

File: .vscode/launch.json (hunk 1 above)
Comment (binnmti) [reply .vscode/launch.json]: Add .vscode/ to gitignore

File: one-pager-maker/src/entity/user/userType.ts (hunk 2 above)
Comment (binnmti) [reply one-pager-maker/src/entity/user/userType.ts]: revert

**Review by Harineko0 — APPROVED** [review]
LGTM!

**Pull request merged**